src/Riok.Mapperly/Descriptors/MappingBuilders/DictionaryMappingBuilder.cs

```diff
@@ -130,4 +133,25 @@ private static (ITypeSymbol, ITypeSymbol)? GetEnumerableKeyValueTypes(MappingBui
 
         return (enumeratedType.TypeArguments[0], enumeratedType.TypeArguments[1]);
     }
+
+    private static INamedTypeSymbol? GetExplicitIndexer(MappingBuilderContext ctx)
+    {
+        if (SymbolEqualityComparer.Default.Equals(ctx.Target.OriginalDefinition, ctx.Types.IDictionaryT))
+            return null;
+
+        // get typed version of IDictionay<>
+        ctx.Target.ImplementsGeneric(ctx.Types.IDictionaryT, out var typedInter);
```

Comment:
Instead of the equals on L139, couldn't we just return null here if `ImplementsGeneric` return false? Event the new overload with the `isExplicit` out param could simplify this a lot.